spring-cloud-gcp-docs/src/main/asciidoc/config.adoc

```diff
@@ -0,0 +1,101 @@
+| `spring.cloud.gcp.config.enabled` | Enables the Config client | `true`
+| `spring.cloud.gcp.config.name` |
+http://cloud.spring.io/spring-cloud-static/spring-cloud-config/1.3.3.RELEASE/single/spring-cloud-config.html#_locating_remote_configuration_resources[Name
+of your application] |
+| `spring.cloud.gcp.config.profile` |
+http://cloud.spring.io/spring-cloud-static/spring-cloud-config/1.3.3.RELEASE/single/spring-cloud-config.html#_locating_remote_configuration_resources[Configuration's
+Spring profile] (e.g., prod) | default
+| `spring.cloud.gcp.config.timeout` | Timeout in milliseconds for connecting to the Google Runtime
+Configuration API | 60000
+| `spring.cloud.gcp.config.project-id` | GCP project ID where the Google Runtime Configuration API
+is hosted, if different from the one in the <<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> |
+| `spring.cloud.gcp.config.credentials.location` | OAuth2 credentials for authenticating with the
+Google Runtime Configuration API, if different from the ones in the
+<<spring-cloud-gcp-core,Spring Cloud GCP Core Module>> |
+| `spring.cloud.gcp.config.credentials.scopes` |
+https://developers.google.com/identity/protocols/googlescopes[OAuth2 scope] for Spring Cloud GCP
+Config credentials | https://www.googleapis.com/auth/cloudruntimeconfig
+|===
+
+NOTE: These properties should be specified in a `bootstrap.yml` file, rather than the usual
+`applications.yml`
+
+=== Quick start
+
+1. Create a configuration in the Google Runtime Configuration API that is called
+`spring.cloud.gcp.config.name`_`spring.cloud.gcp.config.profile`. In other words, if
+`spring.cloud.gcp.config.name` is `myapp` and `spring.cloud.gcp.config.profile` is prod, the
+configuration should be called `myapp_prod`.
++
+In order to do that, you should have the
+https://cloud.google.com/sdk/[Google Cloud SDK] installed, own a Google Cloud Project and run the
+following command:
++
+----
+gcloud beta runtime-config configs create myapp_prod
+gcloud beta runtime-config configs variables set queue_size 25 --config-name myapp_prod
+----
+
+2. Configure your bootstrap.yml file with your application's configuration data:
++
+----
+spring.cloud.gcp.config.name=myapp
+spring.cloud.gcp.config.profile=prod
+----
+3. Add a `@Configuration` class that contains a field annotated with `@Value` in the following way:
++
+----
+@Value("${queue_size}")
+private int queueSize;
+----
+
+When your Spring application starts, if an instance of the class that contains the `queueSize`
+field is auto-wired, the `queueSize` field value will be set to 25.
+
+=== Refreshing the configuration at runtime
+
+Using http://cloud.spring.io/spring-cloud-static/docs/1.0.x/spring-cloud.html#_endpoints[Spring
+Boot Actuator] enables a `/refresh` endpoint in your application that can be used to refresh the
+values of all the Spring Config-managed variables.
+
+To achieve that, add a `@RefreshScope` annotation to your `@Configuration` class(es).
```

Comment:
s/achieve that/achieve this/

Reply:
I think @meltsufin noted that it does n't have to be a `@Configuration` annotation and can be any auto-wired bean ?

Reply:
Ah, touché. I don't know where I read that it had to be a `@Configuration`...

Reply:
Actually, if it isn't an `@Configuration`, the class has to be annotated with something, otherwise it doesn't get picked up... The thing is there might be annotations other than `@Configuration` that also make it possible to auto-wire the class (and not include the `@Configuration` annotation themselves).
I'll reword things here a bit, even though above I say that `@Configuration` should be added, and that should probably stay that way.

Reply:
Some answers should be here: http://cloud.spring.io/spring-cloud-static/Dalston.SR4/multi/multi__spring_cloud_context_application_context_services.html#_refresh_scope

> A Spring `@Bean` that is marked as `@RefreshScope` will get special treatment when there is a configuration change. 

And there is definitely an IMPORTANT not to be cautious with the `@Configuration` classes

Reply:
In my toy app, I'm using a class such as the following one to auto-wire later:

```
@Configuration
public class SampleConfig {

  @Value("${url}")
  private String url;

  public String getUrl() {
    return url;
  }

  public void setUrl(String url) {
    this.url = url;
  }
}
```

Can you recommend an annotation to use other than `@Configuration` that makes beans of that kind auto-wireable?

Reply:
the most generic option is `@Component`

Reply:
... or nothing on the class and in some `@Configuration` have it like:
```
@Bean
@RefreshScope
SampleConfig myConfig() {
   return new SampleConfig();
}
```